qa/rpc-tests/prioritisetransaction.py

```diff
@@ -41,9 +40,17 @@ def run_test (self):
         self.nodes[0].generate(1)
         self.sync_all()
 
-        # Create tx of lower value to be prioritized on node 0
-        # Older transactions get mined first, so this lower value, newer tx is unlikely to be mined without prioritisation
-        priority_tx_0 = self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 0.1)
+        # Create free tx to be prioritized on node 0
+        utxo_list_0 = self.nodes[0].listunspent()
+        assert(len(utxo_list_0) > 0)
+        utxo_0 = utxo_list_0[0]
+        inputs_0 = [{"txid": utxo_0["txid"], "vout": utxo_0["vout"]}]
+        outputs_0 = {self.nodes[1].getnewaddress(): utxo_0["amount"]}
+        raw_tx_0 = self.nodes[0].createrawtransaction(inputs_0, outputs_0)
+        tx_hex_0 = self.nodes[0].signrawtransaction(raw_tx_0)["hex"]
+        priority_tx_0 = self.nodes[0].decoderawtransaction(tx_hex_0)["txid"]
+        print("priority_tx_0: ", priority_tx_0)
+        assert_equal(self.nodes[0].sendrawtransaction(tx_hex_0), priority_tx_0)
```

Comment:
Why is it necessary to create a transaction manually here, instead of using one of the wallet APIs? (I vaguely recall looking at this with you in a pairing, but can't recall the outcome.)

Reply:
i've modified the test to use `sendtoaddress` with `settxfee` to set the initial tx fee to 0.

Reply:
for some reason, `settxfee()` in `rpcwallet.cpp` is not setting the tx fee to 0 as expected. i've reverted to manually creating the transaction.